Property/Typeguard.ts

```diff
@@ -0,0 +1,15 @@
+import { Payload } from "../Payload"
+import { Transformer } from "./Transformer"
+/** Add Tests */
+export class Typeguard<T> implements Transformer {
```

Comment:
Add TODO here if you plan to work on it after this have been merged, else create an issue and remove the comment if it will take a while before you work on it. Also same thing goes for all TODOS in the code.